lib/poller.js

```diff
@@ -98,6 +106,36 @@ class Poller {
     }
   }
 
+  /**
+   * checks if the supplied namespace is allowed to sync the given secret
+   *
+   * @param {Object} namespace namespace object
+   * @param {Object} descriptor secret descriptor
+   */
+  async _isPermitted (namespace, descriptor) {
```

Comment:
Do external secrets users need to add annotations to their namespaces if they're using the assume role feature?

Reply:
they don't have to, it's optional. I added a test to verify all testcases.